ignite/cmd/plugin.go

```diff
@@ -343,6 +347,112 @@ func NewPluginUpdate() *cobra.Command {
 	}
 }
 
+func NewPluginAdd() *cobra.Command {
+	cmdPluginAdd := &cobra.Command{
+		Use:  "add [path]",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a plugin path")
+			}
+			chain, err := newChainWithHomeFlags(cmd)
+			s := cliui.New(cliui.WithStdout(os.Stdout))
+
+			if err != nil {
+				return err
+			}
+			conf, err := chain.Config()
+			if err != nil {
+				return err
+			}
+
+			for _, p := range conf.Plugins {
+				if p.Path == args[0] {
+					return fmt.Errorf("cannot add duplicate plugin %s", args[0])
+				}
+			}
+
+			p := chainconfig.Plugin{
+				Path: args[0],
+			}
+
+			conf.Plugins = append(conf.Plugins, p)
+			s.Printf("🎉 %s added \n", args[0])
+			err = chainconfig.Save(*conf, chain.ConfigPath())
+
+			if err != nil {
+				return err
+			}
+
+			if flag, err := cmd.Flags().GetBool("load"); err == nil && flag {
+				chain, err = newChainWithHomeFlags(cmd)
+
+				if err != nil {
+					return err
+				}
+
+				ctx := context.Background()
+				s.StartSpinner("Loading plugins")
+				plugins, err = plugin.Load(ctx, chain)
```

Comment:
I'm wondering if we shouldn't load the plugin in any case, w/o this `--load` optional flag. WDYT?

Reply:
I was thinking that, I was also thinking if the load operation fails if the newly added plugin should be removed and roll back to the previous state of the config. it seems like the default behavior for most package / plugin managers is to attempt to resolve the new resource and build it if need be. if that is successful then the config will be updated, if not the operation is rolled back and the config definition does not change.

Reply:
Yes, we should do that. First, let's adapt `plugin.Load` so we can load a single plugin. Then call it from there and if there's an error, display it and rollback the config change.

Reply:
ill update to remove the flag and load by default. Now that #3184 has been merged I can add a new `Load` implementation for a single plugin `LoadSingle(ctx context.Context, cfg *pluginsconfig.Config) (*Plugin, error)`